File: 22.c
// Project Euler 22, author: https://github.com/eagletmt/project-euler-c
// I just added the time watching functionality and learned from this beautiful piece :)
/*
 * Copyright (c) 2009, eagletmt
 * Released under the MIT License <http://opensource.org/licenses/mit-license.php>
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <assert.h>
#include <time.h>

static int compare(const void *x, const void *y);

int main(void)
{
  int count = 0;
  FILE *fp;
  char path[] = "names.txt";
  char **names;
  int i;
  int sum = 0;

  clock_t start, end;
  start = clock();

  fp = fopen(path, "r");
  if (!fp) {
    perror(path);
    return 1;
  }

  // 2D malloced array
  names = malloc(8169 * sizeof *names);
  names[count] = malloc(16);
  while (1) {
    fscanf(fp, "\"%[^\"]\",", names[count]);            
    //when the read string was empty
    if(names[count][0] == '\0')
      break;
    names[++count] = malloc(16);
  }
  free(names[count]);
  fclose(fp);

  qsort(names, count, sizeof *names, compare);

  for (i = 0; i < count; i++) {
    int j, s = 0;
    for (j = 0; names[i][j] != 0; j++) {
      s += names[i][j]-'A'+1;
    }
    sum += s * (i+1);

    free(names[i]);
  }
  free(names);

  printf("%d\n", sum);

  end = clock();
  float time_passed = (end-start)*1.0/CLOCKS_PER_SEC;
  printf("\nIt took me: %f seconds.", time_passed); // 0.008291s

  return 0;
}

int compare(const void *x, const void *y)
{
  return strcmp(*(char * const *)x, *(char * const *)y);
}

File: 22.py
#Project Euler solution 22, author: http://github.com/TerkaSlan #
from time import clock

def countValues(size_of_arr):
    # dictionary of letter evaluations
    values = {'A':1,'B':2,'C':3,'D':4,'E':5,'F':6,'G':7,'H':8,'I':9,'J':10, \
          'K':11,'L':12,'M':13,'N':14,'O':15,'P':16,'Q':17,'R':18,'S':19, \
          'T':20,'U':21,'V':22,'W':23,'X':24,'Y':25,'Z':26}
    i=0; score = 0; name_score = 0
    while i<size_of_arr:
        for j in range (0, names[i].__len__(),1):
            name_score += values[names[i][j]]
        score += name_score*i
        name_score = 0
        i+=1
    return score

start = clock()
fp = open('names.txt', 'r')
names = []
names = sorted(fp.read().replace('"', '').split(','), key=str)
end = clock()
print 'The score is: ', countValues(names.__len__()) 
print 'Took me ', end-start ,' secs to come up with the answer' # 0.007076
